src/engine/Map.cxx

```diff
@@ -331,9 +331,7 @@ std::vector<uint8_t> Map::calculateAutotileBitmask(const Point &isoCoordinates)
              mapNodes[it.first * m_columns + it.second]->getMapNodeDataForLayer(currentLayer).tileData &&
              mapNodes[x * m_columns + y] && mapNodes[x * m_columns + y]->getMapNodeDataForLayer(currentLayer).tileData &&
              ((mapNodes[it.first * m_columns + it.second]->getMapNodeDataForLayer(currentLayer).tileID ==
-               mapNodes[x * m_columns + y]->getMapNodeDataForLayer(currentLayer).tileID) ||
-              (mapNodes[it.first * m_columns + it.second]->isLayerAutoTile(currentLayer) &&
-               isLayerAutoTile(isoCoordinates, currentLayer)))))
```

Comment:
this condition must stay because we want to change bitmask orientation for autotiled enabled Tiles
not all tiles are autotiled so we check here in case the current Tile could be autoTiled or not.

are you sure this is the problem ? also did you check it for other types of tiles (something that couldn't be Tiled)?
also @JimmySnails is more familiar than me with the code here.

Reply:
I tested it, seems to work correctly. In line 324 is check whether the current tile is auto-tile. Then in lines 333-334 check whether adjacent tile is same type.

Reply:
we need to check both coordinates, current one and the adjacent one.
in line 324 we check only one of them (current one).
I think if we add an if condition right after the for that check both tiles if they are AutoTiles and then continue with the current if condition that would solve the problem (in my opinion), what do you think ?

Reply:
I think that when current one is autotile and adjacent one is same type, that should imply the adjacent one is also autotile. No additional condition is needed here, and than only autotiles of same type will autotiled.

Reply:
you have a point, makes sense.
I have tested your changes and it's OK.

Thanks.